src/main/scala/com/wavesplatform/state/DataEntry.scala

```diff
@@ -71,7 +76,12 @@ object DataEntry {
                 case JsDefined(JsString(base58)) =>
                   val t = if (base58.isEmpty) Success(Array.emptyByteArray) else Base58.decode(base58) /// Base58 bug
                   t.fold(ex => JsError(ex.getMessage), arr => JsSuccess(BinaryDataEntry(key, ByteStr(arr))))
-                case _ => JsError("value is missing or not a string")
+                case _ => JsError("value is missing or not a base58")
```

Comment:
not-base58 case is handled in line 77, this line handles case when value is not a string, e.g. 8 or true